homeassistant/components/obihai/config_flow.py

```diff
@@ -0,0 +1,86 @@
+"""Config flow to configure the Obihai integration."""
+from __future__ import annotations
+
+from typing import Any
+
+import voluptuous as vol
+
+from homeassistant.config_entries import ConfigFlow
+from homeassistant.const import CONF_HOST, CONF_NAME, CONF_PASSWORD, CONF_USERNAME
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResult
+
+from .connectivity import validate_auth
+from .const import DEFAULT_PASSWORD, DEFAULT_USERNAME, DOMAIN
+
+DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_HOST): str,
+        vol.Optional(
+            CONF_USERNAME,
+            default=DEFAULT_USERNAME,
+        ): str,
+        vol.Optional(
+            CONF_PASSWORD,
+            default=DEFAULT_PASSWORD,
+        ): str,
+    }
+)
+
+
+async def async_validate_credentials(
+    hass: HomeAssistant, user_input: dict[str, Any]
+) -> dict[str, str]:
+    """Manage Obihai options."""
+    errors = {}
+    result = await hass.async_add_executor_job(
+        validate_auth,
+        user_input[CONF_HOST],
+        user_input[CONF_USERNAME],
+        user_input[CONF_PASSWORD],
+    )
+
+    if not result:
+        errors["base"] = "cannot_connect"
+
+    return errors
```

Comment:
I still think this whole method should be simplified to only return a bool, and the errors should be set within the flow.

```suggestion
async def async_validate_credentials(
    hass: HomeAssistant, user_input: dict[str, Any]
) -> bool:
    """Validate Obihai credentials."""
    return await hass.async_add_executor_job(
        validate_auth,
        user_input[CONF_HOST],
        user_input[CONF_USERNAME],
        user_input[CONF_PASSWORD],
    )
```

Reply:
At this stage you have not yet convinced me.
You can always re-adjsut later when the extra entry-points are added and you have a real use case.

Reply:
Guess I don't have much of a choice.  In your court again.